File: algorithms/1441.build-an-array-with-stack-operations/solution.rb
# @param {Integer[]} target
# @param {Integer} n
# @return {String[]}
def build_array(target, n)
  arr = []

  for i in 1..target[-1]
    arr << "Push"
    unless target.include?(i)
      arr << "Pop"
    end
  end

  arr
end

File: algorithms/1441.build-an-array-with-stack-operations/solution.rs
impl Solution {
    pub fn build_array(target: Vec<i32>, n: i32) -> Vec<String> {
        let mut arr: Vec<String> = vec![];
        let last = *target.last().unwrap();

        for i in 1..=last {
            arr.push("Push".to_string());
            if !target.contains(&i) {
                arr.push("Pop".to_string());
            }
        }

        arr
    }
}
